### PR TITLE
refactor nft with cosmos-sdk nft module

### PR DESCRIPTION
**Pull request by dreamer-zq**

### CODE REVIEW

**Review by taramakage — APPROVED** [review]

File: modules/nft/module/module.go [hunk 1]
```diff
@@ -136,18 +138,18 @@ func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, data json.
...
+	gs := am.keeper.ExportGenesis(ctx)
 	return cdc.MustMarshalJSON(gs)
 }
 
 // ConsensusVersion implements AppModule/ConsensusVersion.
-func (AppModule) ConsensusVersion() uint64 { return 1 }
+func (AppModule) ConsensusVersion() uint64 { return 2 }
 
 // BeginBlock performs a no-op.
 func (AppModule) BeginBlock(_ sdk.Context, _ abci.RequestBeginBlock) {}
```
Comment (taramakage) [inline modules/nft/module/module.go:155]: Dead BeginBlock and EndBlock can be removed. See [#12603](https://github.com/cosmos/cosmos-sdk/pull/12603).

File: modules/nft/keeper/collection.go [hunk 2]
```diff
@@ -34,85 +27,42 @@ func (k Keeper) SetCollection(ctx sdk.Context, collection types.Collection) erro
...
-		return nil
-	})
+	nfts, err := k.GetNFTs(ctx, denomID)
 	if err != nil {
-		return types.Collection{}, nil, status.Errorf(codes.InvalidArgument, "paginate: %v", err)
+		return types.Collection{}, err
 	}
-	return types.NewCollection(denom, nfts), pageRes, nil
+	return types.NewCollection(*denom, nfts), nil
 }
```
Comment (taramakage) [inline modules/nft/keeper/collection.go:41]: Is `GetCollection` necessary anymore since the REST query has been deprecated?

Comment (taramakage) [reply modules/nft/keeper/collection.go:41]: `GetPaginateCollection` was only called by `Collection` in gRPC query, so we don't need it anymore?

Comment (dreamer-zq) [reply modules/nft/keeper/collection.go:41]: > Is `GetCollection` necessary anymore since the REST query has been deprecated?

deleted

Comment (dreamer-zq) [reply modules/nft/keeper/collection.go:41]: > `GetPaginateCollection` was only called by `Collection` in gRPC query, so we don't need it anymore?

yes

**Review by aofengli — APPROVED** [review]

**Pull request merged**